tests/validation/src/jacobian_validation.cpp

```diff
@@ -0,0 +1,1475 @@
+    setup_csv_header_covariance(rect_cov_transport_file);
+
+    std::ofstream wire_cov_transport_file;
+    wire_cov_transport_file.open(path + "wire_cov_transport.csv");
+    setup_csv_header_covariance(wire_cov_transport_file);
+
+    // Output Csv file (RK tolerance iteration mode)
+    std::vector<std::ofstream> rect_files(log10_tols.size());
+    std::vector<std::ofstream> wire_files(log10_tols.size());
+
+    if (rk_tolerance_iterate_mode) {
+        for (std::size_t i = 0u; i < log10_tols.size(); i++) {
+            const std::string rect_name = "inhom_rect_material_" +
+                                          std::to_string(int(log10_tols[i])) +
+                                          ".csv";
+            const std::string wire_name = "inhom_wire_material_" +
+                                          std::to_string(int(log10_tols[i])) +
+                                          ".csv";
+            rect_files[i].open(path + rect_name);
+            wire_files[i].open(path + wire_name);
+
+            setup_csv_header_jacobian(rect_files[i]);
+            setup_csv_header_jacobian(wire_files[i]);
+        }
+    }
+
+    // Memory resource
+    vecmem::host_memory_resource host_mr;
+
+    // Filter out the google test flags
+    ::testing::InitGoogleTest(&argc, argv);
+
+    // Detector types
+    using rectangle_telescope = detector<telescope_metadata<rectangle2D<>>>;
+    using wire_telescope = detector<telescope_metadata<line<true>>>;
+
+    mask<rectangle2D<>> rect{0u, 10000.f * unit<scalar>::mm,
+                             10000.f * unit<scalar>::mm};
+    mask<line<true>> wire{0u, 10000.f * unit<scalar>::mm,
+                          10000.f * unit<scalar>::mm};
+
+    using track_type = free_track_parameters<transform3_type>;
+
+    // Constant magnetic field type
+    using const_bfield_t = bfield::const_field_t;
+
+    // Magnetic field map using nearest neightbor interpolation
+    using inhom_bfield_t = bfield::inhom_field_t;
+
+    const const_bfield_t const_bfield = bfield::create_const_field(B_z);
+    const inhom_bfield_t inhom_bfield = bfield::create_inhom_field();
+
+    // Actor chain type
+    using actor_chain_t = actor_chain<dtuple, propagation::print_inspector,
+                                      parameter_transporter<transform3_type>,
+                                      bound_getter<transform3_type>,
+                                      parameter_resetter<transform3_type>>;
+
+    // Iterate over reference (pilot) tracks for a rectangular telescope
+    // geometry and Jacobian calculation
+    using uniform_gen_t =
+        random_numbers<scalar, std::uniform_real_distribution<scalar>,
+                       std::seed_seq>;
+    using trk_generator_t = random_track_generator<track_type, uniform_gen_t>;
+    trk_generator_t::configuration trk_gen_cfg{};
+    trk_gen_cfg.n_tracks(n_tracks + n_skips);
+    trk_gen_cfg.phi_range(-constant<scalar>::pi + 0.1f,
```

Comment:
In case you were worried about the poles at pi, the track generator already adds an epsilon there I think

Reply:
OK